engine/jobmaster/dm/task_manager.go

```diff
@@ -282,7 +282,14 @@ func (tm *TaskManager) allFinished(ctx context.Context) bool {
 			return false
 		}
 		// update if we add new task mode
-		if runningTask.Unit != frameModel.WorkerDMLoad || task.Cfg.TaskMode != dmconfig.ModeFull {
+		switch task.Cfg.TaskMode {
+		case dmconfig.ModeFull:
+			if runningTask.Unit != frameModel.WorkerDMLoad {
+				return false
+			}
+		case dmconfig.ModeDump:
```

Comment:
in old behaviour, we don't return true before finish checking all `job.Tasks`, so I'm not sure we should directly `return true` here